src/IO/S3/copyS3File.cpp

```diff
@@ -643,7 +707,7 @@ namespace
 
             if (!outcome.IsSuccess())
             {
-                abortMultipartUpload();
+                scheduleAbortMultipartUpload();
```

Comment:
Alternative here is to wait the rest of the tasks. Log their exceptions. Propagate here the first exception.

Reply:
It makes sense. But it could be implemented in a more simple way. Like set atomic bool when exception has happen. And check this flag when new task is being scheduled or it starts to run in thread pool.

However in real world that efforts would be spend in vain: all others task most like is blocked on socket operations and do not check if there is exception has happen in other threads.
So that could be implemented, but this should not be the main purpose of the code because the real gain is small.

Reply:
I implemented this idea and reverted my previous commits.
I hope to backport this PR, since the current issue obscures the real problem we've been debugging: https://clickhouse-inc.slack.com/archives/C03JLG3K7QA/p1754935130959249 cc @mlex.
Switching to ThreadPoolTaskTracker.h would be nice for master, but it might be too much for a backport.

I also like how https://github.com/ClickHouse/ClickHouse/pull/86764 looks: the code is much cleaner, and it’s nice that the task tracker encapsulates the switch between the initialized and uninitialized scheduler.